packages/imperative/src/plugins/PluginManagementFacility.ts

```diff
@@ -1260,4 +1263,17 @@ export class PluginManagementFacility {
             }
         }
     }
+
+    private getPluginName(plugin: string | false | ICredentialManager, setting: string): string | boolean {
+        if(ImperativeConfig.instance.isCredentialManager(plugin)) {
```

Comment:
We could give this a try, I guess when I originally wrote this I wanted to make sure that when passing in an object from the imperative override we can have always the right one and not attempt to work with an object that does not satisfy the interface

Reply:
Added a check instead for whether the value passed in is truthy this should evaluate for `null`, `undefined` or `false` since we care about it being an object that has value inside of them as well